fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/mapper/loan/PartialEnricherSupport.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.event.external.service.serialization.mapper.loan;
+
+import org.apache.fineract.infrastructure.core.service.PartialDataEnricherProcessor;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Context;
+import org.mapstruct.MappingTarget;
+
+public interface PartialEnricherSupport {
+
+    @AfterMapping
+    default <T> void enrich(@MappingTarget final T target, @Context PartialDataEnricherProcessor processor) {
```

Comment:
I'm not sure I'm a fan of wiring this into the mapstruct mapping cause the serializers don't enforce the usage of mapstruct.

I think we should hook this up directly in the event serializers.

Reply:
removed because of the BusinessEventSerializer changes